File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    let n = nums.length / 2;
    let hash = {}
    
    nums.map((a) => {
        if(!(hash[a])) hash[a] = 1;
        else hash[a] += 1
    })
    
    let m = Object.keys(hash).length;
    
    if(m > n) return n;
    if(m <= n) return m;
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        int num = nums.length / 2;
        
        HashSet<Integer> hash = new HashSet<>();
        
        for(int n : nums) {
            hash.add(n);
        }
        
        if(hash.size() > num) {
            return num;
        }
        else return hash.size();
    }
}